### PR TITLE
(MAINT) Make Splunk Releasable

### PR DESCRIPTION
**Pull request by RandomNoun7**
A lot of proof of concept work for sending Activities API events from
PE to Splunk was done on the main branch of this repo. There has been
a major change in strategy such that this work was no longer valid.

Unfortunately there we some contributions and bug fixes to this module
that got interwoven with the Events sending code. We can't release the
module with all of the dead event sending code still in it, but that
also meant that we could not release the legitimate bug fixes and
improvements.

This PR removes all of the Events sending code while retaining the
interwoven bug fixes and improvements, without re-writing the history
of the main branch so that the module can be brought to a releasable
state.

# Summary

# Detailed Description

# Checklist

[ ] Draft PR?
[ ] Ensure README is updated
  [ ] Any changes to existing documentation
  [ ] Anything new added
  [ ] Link to external Puppet documentation
[ ] Tags
[ ] Unit Tests
[ ] Acceptance Tests
[ ] PR title is "(Ticket|Maint) Short Description"
[ ] Commit title matches PR title

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,7 +61,7 @@ Instructions assume you are using Puppet Enterprise. For Open Source Puppet inst
   * Click **Review** and then **Submit**.
   * When complete the HEC token should look something like this:
 
-  ![hec_token](https://raw.githubusercontent.com/puppetlabs/puppetlabs-splunk_hec/v0.8.1/docs/images/hec_token.png)
+  ![hec_token](https://raw.githubusercontent.com/puppetlabs/puppetlabs-splunk_hec/main/docs/images/hec_token.png)
```
Comment (RandomNoun7) [inline README.md:64]: There's a bit of chicken and egg with these urls. We might have to do something like update these urls to refer to the next version tag before we actually create the tag, and then tag a sha with the release version soon thereafter to make the urls work?

Comment (RandomNoun7) [reply README.md:64]: I updated this to main for instance, but I think I will revert that change and we can deal with this problem later.

File: manifests/init.pp [hunk 2]
```diff
@@ -136,7 +98,7 @@
     }
   }
 
-  file { "${settings::confdir}/splunk_hec.yaml":
+  file { '/etc/puppetlabs/puppet/splunk_hec.yaml':
```
Comment (gsparks) [inline manifests/init.pp:101]: Is this part of common events? It seems like basing the config file off the confdir is good either way.

**Review by gsparks — DISMISSED** [review]
Just a typo. Looks like you got everything though!

File: docs/custom_installation.md [hunk 3]
```diff
@@ -20,20 +20,20 @@ The steps below will help install and troubleshoot the report processor on a sin
...
 7. If configured properly the Puppet Report Viewer app in Splunk will show 1 node in the Overview tab.
 
-8. Now it is time to roll these settings out to the fleet of the Puppet Masters in the installation. For Puppet Enterprise users:
+8. Now it is time to roll these settings out to the fleet of to the Puppet Masters in the installation. For Puppet Enterprise users:
 	- Navigate to Classification -> PE Infrastructure -> PE Master
 	- Select Configuration
 	- Press Refresh to ensure the splunk_hec class is loaded
 	- Add new class `splunk_hec`
-	- From the `Parameter name` select at least `url` and `splunk_token` and provide the same attributes from the testing configuration file
+	- From the `Parameter name` select atleast `url` and `token` and provide the same attributes from the testing configuration file
```
[the remark this answers is not in the document]
Comment (RandomNoun7) [reply docs/custom_installation.md]: Nice. Done.

File: manifests/init.pp (hunk 2 above)
Comment (RandomNoun7) [reply manifests/init.pp]: Yeah, we can do this here.

**Review by nam054 — APPROVED** [review]

**Pull request merged**